Cursor/funkybot_Mouse Buddy.lua

```diff
@@ -0,0 +1,805 @@
+-----------------------------------------------------------
+local ManualModeDefault = false
+local mode = "live"
+if ManualModeDefault then mode = "manual" end
+
+-----------------------------------------------------------
+-- Global variables for HTML file path (persistent state)
+-----------------------------------------------------------
+local EXT_SECTION = "MM_BUDDY"
+local EXT_KEY = "HTML_PATH"
+local DEFAULT_PATH = " "  -- default if not set
+
+local HTML_PATH = r.GetExtState(EXT_SECTION, EXT_KEY)
+if HTML_PATH == "" then
+  HTML_PATH = DEFAULT_PATH
+  r.SetExtState(EXT_SECTION, EXT_KEY, HTML_PATH, true)
+end
+local pathInput = HTML_PATH
+
+-----------------------------------------------------------
+-- Mode selection variables (for Manual Mode dropdowns)
+-----------------------------------------------------------
+local selectedManualIndex = 0    -- 0-based index; 0 corresponds to "All"
+local selectedBehaviorIndex = 0  -- 0-based index
+
+-----------------------------------------------------------
+-- Global variable to store the last valid live context.
+-----------------------------------------------------------
+local lastValidContext = nil
+
+-----------------------------------------------------------
+-- Global variables for freezing the display (including segment)
+-----------------------------------------------------------
+local freezeDisplay = false    -- when true, the live view will be frozen
+local frozenContext = "UNKNOWN"  -- will store the live context (window) when frozen
+local frozenSegment = "UNKNOWN"  -- will store the live segment when frozen
+local frozenDetails = ""         -- will store the live details when frozen
+
+-----------------------------------------------------------
+-- Predefined valid context groups and mouse filter options
+-----------------------------------------------------------
+local validContextGroups = {
+  "Arrange View",
+  "Edit Cursor Handle",
+  "Envelope Control Panel",
+  "Envelope Point",
+  "Envelope Segment",
+  "Fixed Lane Comp area",
+  "MIDI CC Event",
+  "MIDI CC Lane",
+  "MIDI CC Segment",
+  "MIDI Editor",
+  "MIDI Note",
+  "MIDI Piano Roll",
+  "MIDI Ruler",
+  "MIDI Source Loop end Marker",
+  "Media Item Bottom Half",
+  "Media Item",
+  "Media Item Edge",
+  "Media Item Fade/Autocrossfade",
+  "Media Item Stretch Marker",
+  "Media Item Take Marker",
+  "Mixer Control Panel",
+  "Project Tempo/Time Signature Marker",
+  "Project marker/region",
+  "Razor Edit Area",
+  "Razor Edit Edge",
+  "Razor Edit Envelope",
+  "Ruler"
+}
+
+local validMouseFilters = {
+  "Left Click",
+  "Left Click/Drag",
+  "Left Drag",
+  "Right Click",
+  "Double Click",
+  "Right Drag",
+  "Middle Click",
+  "Middle Drag"
+}
+
+-----------------------------------------------------------
+-- Create an ImGui context with docking enabled.
+-----------------------------------------------------------
+local ctx = r.ImGui_CreateContext("Mouse Buddy 3.0", r.ImGui_ConfigFlags_DockingEnable())
```

Comment:
1. Publicly released ReaImGui scripts must enable backward compatibility by specifying which version they are written for. See snippets at https://forum.cockos.com/showthread.php?t=250419.
  The ImGui.* (instead of r.ImGui_) convention is optional but recommended for enhanced backward compat. More details & script to auto-convert existing code at https://forum.cockos.com/showpost.php?p=2775338.
3. `ConfigFlags_DockingEnable` has been enabled by default for a few years. Explicitly specifying it here overrides the user's corresponding setting. (If that is the intention, then that is fine.)